From 19314f744d833b402dcaedd358cc6ed385154733 Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 29 Jun 2022 11:01:50 +0300
Subject: [PATCH 11/33] qspi: env: Remove obsolete flash environment offsets

These are not needed when using mtd qspi partitions approach.

Issue: ALB-9014
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32-cc.h | 47 +---------------------------------------
 1 file changed, 1 insertion(+), 46 deletions(-)

diff --git a/include/configs/s32-cc.h b/include/configs/s32-cc.h
index 1dc2f598a0..b709d85150 100644
--- a/include/configs/s32-cc.h
+++ b/include/configs/s32-cc.h
@@ -77,42 +77,6 @@
 /* Increase image size */
 #define CONFIG_SYS_BOOTM_LEN		(SZ_64M)
 
-/**
- * Note: The *_FLASH_ADDR and *_FLASH_MAXSIZE macros are used
- * with the 'setexpr' command. Therefore ensure none of them expand
- * into operations with more than two operands and avoid unnecessary
- * parantheses. Also these should be kept in sync with
- * 'conf/machine/include/s32*flashmap.inc'.
- *
- * QSPI flash map:
- *
- * Name		Size			Offset
- * FIP		~1.9M			0x0
- * QSPI env	64K(CONFIG_ENV_SIZE)	0x01e0000(CONFIG_ENV_OFFSET)
- * Image	14M			0x0e00000
- * Linux DTB	1M			0x0ff0000
- * Ramdisk	32M			0x10f0000
- */
-#define KERNEL_FLASH_MAXSIZE		0x0e00000
-#define FDT_FLASH_MAXSIZE		0x0100000
-#define RAMDISK_FLASH_MAXSIZE		0x2000000
-#define FIP_FLASH_ADDR			0x0000000
-#define KERNEL_FLASH_ADDR		0x01f0000
-#define FDT_FLASH_ADDR			0x0ff0000
-#define RAMDISK_FLASH_ADDR		0x10f0000
-
-#if defined(CONFIG_ENV_IS_IN_SPI_FLASH)
-#  if (CONFIG_ENV_OFFSET + CONFIG_ENV_SIZE > KERNEL_FLASH_ADDR)
-#    error "Environment and Kernel would overlap in flash memory"
-#  endif
-#endif
-#if (KERNEL_FLASH_ADDR + KERNEL_FLASH_MAXSIZE > FDT_FLASH_ADDR)
-#  error "Kernel and FDT would overlap in flash memory"
-#endif
-#if (FDT_FLASH_ADDR + FDT_FLASH_MAXSIZE > RAMDISK_FLASH_ADDR)
-#  error "FDT and Ramdisk would overlap in flash memory"
-#endif
-
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + SZ_2M)
 
@@ -183,16 +147,7 @@
 		"${boot_mtd} ${loadaddr} ${ramdisk_addr} ${fdt_addr};\0" \
 	"flashbootargs=setenv bootargs console=${console},${baudrate}" \
 		" root=/dev/ram rw earlycon " EXTRA_BOOT_ARGS ";"\
-		"setenv fip_flashaddr " __stringify(FIP_FLASH_ADDR) ";" \
-		"setenv kernel_flashaddr " __stringify(KERNEL_FLASH_ADDR) ";" \
-		"setenv flashsize " __stringify(FSL_QSPI_FLASH_SIZE) ";" \
-		"setenv kernel_maxsize " __stringify(KERNEL_FLASH_MAXSIZE) ";" \
-		"setenv fdt_flashaddr " __stringify(FDT_FLASH_ADDR) ";" \
-		"setenv fdt_maxsize " __stringify(FDT_FLASH_MAXSIZE) ";" \
-		"setenv ramdisk_flashaddr " \
-				__stringify(RAMDISK_FLASH_ADDR) ";" \
-		"setenv ramdisk_maxsize " \
-				__stringify(RAMDISK_FLASH_MAXSIZE) ";\0" \
+		"setenv flashsize " __stringify(FSL_QSPI_FLASH_SIZE) ";\0" \
 	"image=Image\0" \
 	"initrd_high=" __stringify(S32CC_INITRD_HIGH_ADDR) "\0" \
 	"ipaddr=" S32CC_DEFAULT_IP "\0"\
-- 
2.17.1

